demos/object_detection_demo/jupyter-python/object_detection_demo.ipynb

```diff
@@ -0,0 +1,750 @@
+    "    )\n",
+    "    start_time = perf_counter()\n",
+    "    inference_result = do_inference_on_video(\n",
+    "        detector_pipeline=detector_pipeline, input_filename=input_filename, jump_frames=JUMP_FRAMES\n",
+    "    )\n",
+    "    end_time = perf_counter()\n",
+    "\n",
+    "    # Draw inference results on video frames and compute FPS\n",
+    "    has_landmarks = architecture_type == \"retina\"\n",
+    "    result_frames = make_result_frames(inference_result=inference_result, has_landmarks=has_landmarks)\n",
+    "    fps = len(result_frames) / (end_time - start_time)\n",
+    "\n",
+    "    return result_frames, fps"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "certified-browse",
+   "metadata": {},
+   "source": [
+    "## Create widgets\n",
+    "\n",
+    "This demo works with a variety of [Open Model Zoo](https://github.com/openvinotoolkit/open_model_zoo/) models and allows you to use your own video.  We create widgets with [IPywidgets](https://github.com/jupyter-widgets/ipywidgets) to easily select a demo video or choose a video from your PC.\n",
+    "\n",
+    "Sample videos are downloaded from [Intel IoT Sample Videos](https://github.com/intel-iot-devkit/sample-videos). The link shows previews of the videos. If you upload your own video, it is recommended to use a short video. The maximum video size is 10MB."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "creative-appreciation",
+   "metadata": {},
+   "source": [
+    "## Download or upload a video"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "aggressive-attitude",
+   "metadata": {},
+   "source": [
+    "### Option 1: Download a sample video"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "established-reynolds",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sample_video_base_url = \"https://github.com/intel-iot-devkit/sample-videos/raw/master\"\n",
+    "sample_video_filenames = open(\"sample_videos.lst\").read().splitlines()\n",
+    "sample_video_list = [(fn[:-4], f\"{sample_video_base_url}/{fn}\") for fn in sample_video_filenames]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "appreciated-boundary",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sample_video = widgets.Dropdown(options=sample_video_list, index=5)\n",
+    "sample_video"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "empirical-median",
+   "metadata": {},
+   "source": [
+    "### Option 2: Upload your own video"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "technical-grocery",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import time\n",
+    "start = time.perf_counter()\n",
+    "uploader = widgets.FileUpload(multiple=False)\n",
+    "uploader"
```

Comment:
Please move embedding of sample_video and uploader down to show_results_on_model and show_inference_mulitple_models.
I.e. delete ",sample_video" and ",uploader" here
and add something like this:
    videos = sample_video,
    custom_video = uploader
to show_results_on_model and show_inference_mulitple_models functions as arguments.
You'll probably have to modify get_input_filename() function as well to pass instances of these 2 widgets as arguments.
Main goal is to keep all widgets in one place - where actually inference is done and results are shown.

Reply:
That is nicer indeed, I agree that the input video and especially the second inference function are too far away. I find that if I add them with the inference it adds confusion though. It is not clear that the uploader widget is optional. 

I could also delete the uploader widget, and include video's that are uploaded to the notebook in (what is now the) the sample video dropdown. I would add a line on how to do that, so that adds a bit of complexity, but it reduces UI clutter a lot and makes the code easier to read (also getting rid of the `get_input_filename` function that is a bit of a clutch). Downside is that it doesn't work in the "app version" with Voila, but that is not a priority. What do you think? Other suggestions welcome too.

Reply:
We can just put label "Optional" near Upload.
Another option is to add labels, something like:
Select preloaded video from dropdown: [videos list]
- or - 
Upload your own video: [Upload button]

Not sure if it would be easy to do using currently used functions/decorators.

BTW, in case of standalone application current version is fine, but in case of notebook videos list and upload  are too far from actual inference

Reply:
I changed this in the most recent commit. I now only show the dropdown, which now shows sample video's and user's own video's. There is a small drawback that the user now has to use Jupyter's upload function instead of a widget, but I think it looks much clearer now and there is no awkward `get_input_filename` function anymore.